src/Facade/Services/PurchaseOrderDeliveryFacadeService.cs

```diff
@@ -0,0 +1,87 @@
+﻿namespace Linn.Purchasing.Facade.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    using Linn.Common.Facade;
+    using Linn.Common.Persistence;
+    using Linn.Purchasing.Domain.LinnApps;
+    using Linn.Purchasing.Domain.LinnApps.Keys;
+    using Linn.Purchasing.Domain.LinnApps.PurchaseOrders;
+    using Linn.Purchasing.Resources;
+    using Linn.Purchasing.Resources.RequestResources;
+
+    public class PurchaseOrderDeliveryFacadeService : IPurchaseOrderDeliveryFacadeService
+    {
+        private readonly IRepository<PurchaseOrderDelivery, PurchaseOrderDeliveryKey> repository;
+
+        private readonly IBuilder<PurchaseOrderDelivery> resourceBuilder;
+
+        private readonly ITransactionManager transactionManager;
+
+        private readonly IPurchaseOrderDeliveryService domainService;
+
+        public PurchaseOrderDeliveryFacadeService(
+            IRepository<PurchaseOrderDelivery, PurchaseOrderDeliveryKey> repository,
+            IBuilder<PurchaseOrderDelivery> resourceBuilder,
+            IPurchaseOrderDeliveryService domainService,
+            ITransactionManager transactionManager)
+        {
+            this.repository = repository;
+            this.resourceBuilder = resourceBuilder;
+            this.domainService = domainService;
+            this.transactionManager = transactionManager;
+        }
+
+        public IResult<IEnumerable<PurchaseOrderDeliveryResource>> SearchDeliveries(
+            string supplierSearchTerm, string orderNumberSearchTerm, bool includeAcknowledged)
+        {
+            var results = this.domainService.SearchDeliveries(
+                supplierSearchTerm,
+                orderNumberSearchTerm,
+                includeAcknowledged);
+            return new SuccessResult<IEnumerable<PurchaseOrderDeliveryResource>>(
+                results.Select(x => (PurchaseOrderDeliveryResource)this.resourceBuilder.Build(x, null)));
+        }
+
+        public IResult<PurchaseOrderDeliveryResource> PatchDelivery(
+            PurchaseOrderDeliveryKey key,
+            PatchRequestResource<PurchaseOrderDeliveryResource> requestResource, 
+            IEnumerable<string> privileges)
+        {
+            var entity = this.repository.FindById(key);
+
+            if (requestResource.From.DateAdvised != requestResource.To.DateAdvised)
```

Comment:
would it be safer for these to compare entity.DateAdvised etc to the resource in this method?

Reply:
That's a good question. 

In my head the person requesting the PATCH defines a 'from' and a 'to' state and that's what's used to decide whether to apply an update to any given field or not. 

So I think comparing the two things in the resource is probably correct.. Feel free to disagree though

Reply:
Yeah fair enough, I was just thinking what if we accidentally altered the 'from' state before sending it to the backend here, but actually I can't really see a scenario where it matters.. Unless it happened to be the same as the new To and so didn't update maybe?

Reply:
But yeah either using part of the resource or the entity as the from state will be fine I'm sure, and if both in resource seems more pure then go for it

Reply:
I'll earmark this for the next show & tell and we can discuss it before I think about adding it to the Common code